From 560fba93c4c1a6c7475ee33e1d7a1a840e5fbc7a Mon Sep 17 00:00:00 2001
From: Thomas Bartholomae <t.bartholomae@intel.com>
Date: Thu, 3 Sep 2020 18:53:05 +0200
Subject: [PATCH] PMCPESYS-245: used err codes in valid temp range

in case of error condition standard error codes are used e.g. -EIO,
-EINVAL. These error codes are inside the valid temperature range,
therefore we can't use it. An individual ETEMPINVAL = -255 is defined
instead, which is outside the valid temperature range.
---
 drivers/thermal/ltq_thermal.c | 37 ++++++++++++++-----------------------
 1 file changed, 14 insertions(+), 23 deletions(-)

diff --git a/drivers/thermal/ltq_thermal.c b/drivers/thermal/ltq_thermal.c
index 3e4575152318..9e17fedd1350 100644
--- a/drivers/thermal/ltq_thermal.c
+++ b/drivers/thermal/ltq_thermal.c
@@ -39,11 +39,8 @@ struct ltq_thermal_sensor_ops {
 
 struct ltq_thermal_sensor {
 	int id;
-
 	struct thermal_zone_device *tzd;
-
 	const struct ltq_thermal_sensor_ops *ops;
-
 	int temp;
 	int last_temp;
 	int emul_temp;
@@ -82,21 +79,21 @@ struct ltq_thermal {
 #define INT_LVL			0xFFFF0000
 
 /* Temp sensor data (read out) register */
-#define	TS_CODE		0x00000FFF
-#define	TS_DV		0x80000000
+#define	TS_CODE			0x00000FFF
+#define	TS_DV			0x80000000
 
-#define TS_REG_OFF			0x10U
-#define INT_LVL_POS			16
+#define TS_REG_OFF		0x10U
+#define INT_LVL_POS		16
 #define PRX300_TS_CODE		0x3FFu
 #define VSAMPLE_MASK		0x4
-#define VSAMPLE_V			0x4
-#define VSMPLE_NV			0x0
+#define VSAMPLE_V		0x4
+#define VSMPLE_NV		0x0
 #define CTRL_REG_ID(__X__)	(CTRL_REG + ((__X__) * TS_REG_OFF))
 #define DATA_REG_ID(__X__)	(DATA_REG + ((__X__) * TS_REG_OFF))
-#define TIMEOUT				1000
-#define ETEMPINVAL			-255
+#define TIMEOUT			1000
+#define ETEMPINVAL		255
 
-void ltq_prx300_init(struct ltq_thermal_sensor *sensor)
+static void ltq_prx300_init(struct ltq_thermal_sensor *sensor)
 {
 	struct ltq_thermal *priv = sensor->drvdata;
 
@@ -107,7 +104,7 @@ static int ltq_prx300_get_temp(struct ltq_thermal_sensor *sensor)
 {
 	struct ltq_thermal *priv = sensor->drvdata;
 	u32 val;
-	int ts_val = ETEMPINVAL;
+	int ts_val = -ETEMPINVAL;
 	int a0 = -49;
 	int v1 = 10;
 	int v2 = 1000;
@@ -149,7 +146,7 @@ static int ltq_prx300_get_temp(struct ltq_thermal_sensor *sensor)
 	return (ts_val / 1000); /* returns °mC */
 }
 
-void  ltq_grx500_init(struct ltq_thermal_sensor *sensor)
+static void  ltq_grx500_init(struct ltq_thermal_sensor *sensor)
 {
 	struct ltq_thermal *priv = sensor->drvdata;
 
@@ -175,7 +172,7 @@ void  ltq_grx500_init(struct ltq_thermal_sensor *sensor)
 			   SOC_NC | TS_EN_SHUNT);
 }
 
-int ltq_grx500_get_temp(struct ltq_thermal_sensor *sensor)
+static int ltq_grx500_get_temp(struct ltq_thermal_sensor *sensor)
 {
 	u32 reg;
 	int T, v;
@@ -186,7 +183,7 @@ int ltq_grx500_get_temp(struct ltq_thermal_sensor *sensor)
 	struct ltq_thermal *priv = sensor->drvdata;
 
 	if (!priv)
-		return -EINVAL;
+		return -ETEMPINVAL;
 
 	/* Select a channel */
 	regmap_update_bits(priv->chiptop, CTRL_REG, CH_SEL_MASK,
@@ -205,7 +202,7 @@ int ltq_grx500_get_temp(struct ltq_thermal_sensor *sensor)
 			   SOC_NC | TS_EN_SHUNT);
 
 	if (ret)
-		return -EIO;  /* temp read timeouted */
+		return -ETEMPINVAL;  /* temp read timeout */
 
 	v = reg & TS_CODE;
 
@@ -272,12 +269,6 @@ static int ltq_thermal_get_trend(void *data, int trip,
 		*trend = THERMAL_TREND_RAISING;
 	else if (temp < last_temp && temp <= (trip_temp - trip_hyst))
 		*trend = THERMAL_TREND_DROPPING;
-/*
-	else if (temp < last_temp && temp > (trip_temp - trip_hyst))
-		*trend = THERMAL_TREND_DROPPING;
-	else if (temp < last_temp && temp <= (trip_temp - trip_hyst))
-		*trend = THERMAL_TREND_DROP_FULL;
-*/
 	else
 		*trend = THERMAL_TREND_STABLE;
 
